backend/test/akvo/lumen/visualisation/map_test.clj

```diff
@@ -0,0 +1,177 @@
+   [akvo.lumen.fixtures :refer [*error-tracker*
+                                *tenant-conn*
+                                data-groups-future-fixture
+                                error-tracker-fixture
+                                system-fixture
+                                tenant-conn-fixture]]
+   [akvo.lumen.lib.visualisation.map-config :as v.map-config]
+   [akvo.lumen.lib.visualisation.maps :as v.maps]
+   [akvo.lumen.db.transformation :as db.transformation]
+   [akvo.lumen.specs.import :as i-c]
+   [akvo.lumen.test-utils :as tu]
+   [clj-time.format :as f]
+   [clojure.java.jdbc :as jdbc]
+   [clojure.walk :as walk]
+   [clojure.test :refer [deftest testing is use-fixtures]]))
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Groups definition, the form of generated data is later used in utility fns
+;;; where columns that for example makes sense as pointColorcolumn is picked
+;;; based on the form of groups-definition
+
+(def groups-def [{:groupId "group1"
+                  :groupName "repeatable group"
+                  :repeatable true
+                  :column-types ["text" "number" "geopoint"]
+                  :max-responses 10}
+                 {:groupId "group2"
+                  :groupName "not repeatable group"
+                  :repeatable false
+                  :column-types ["number" "date"]}])
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Fixture
+;;;
+
+(def ^:dynamic *dataset-data*)
+(defn dataset-data-fixture [f]
+  (binding [*dataset-data* (i-c/flow-sample-imported-dataset groups-def 2)]
+    (f)))
+
+(def ^:dynamic *dataset-id*)
+(defn dataset-id-fixture [f]
+  (binding [*dataset-id* (tu/import-file *tenant-conn* *error-tracker*
+                                         {:dataset-name "Generated map"
+                                          :kind "clj-flow"
+                                          :data *dataset-data*})]
+    (f)
+    (let [{:keys [imported-table-name table-name]}
+          (db.transformation/latest-dataset-version-by-dataset-id
+           *tenant-conn* {:dataset-id *dataset-id*})
+          drop-table-seq
+          (format "DROP SEQUENCE IF EXISTS %s_rnum_seq CASCADE" table-name)
+          drop-table (jdbc/drop-table-ddl table-name)
+          drop-imported-table-seq
+          (format "DROP SEQUENCE IF EXISTS %s_rnum_seq CASCADE" imported-table-name)
+          drop-imported-table (jdbc/drop-table-ddl imported-table-name)]
+      (jdbc/db-do-commands *tenant-conn* [drop-table-seq
+                                          drop-table
+                                          drop-imported-table-seq
+                                          drop-imported-table])
+      (dataset/delete *tenant-conn* *dataset-id*))))
+
+
+(use-fixtures :once
+  system-fixture data-groups-future-fixture tenant-conn-fixture
+  error-tracker-fixture tu/spec-instrument dataset-data-fixture
```

Comment:
I'm interested to hear why we're using *fixtures* for setting up the test? 🤔 
Other functional tests prepare the dataset/visualization/etc as part of the test and *not* in a fixture. Are we going to reuse them? If so, is not the `akvo.lumen.fixtures` the place for this code?

Reply:
The thought was to prepare the dataset, write on test for existing maps, and then clean up the data after the test is done. This could be handled without the fixture but since I knew we wanted to write a test to verify the functionality between current maps and the future implementation of dataset-version-2 maps I put it in a fixture so we can do another deftest but use the same data. 

This said, maybe fixtures should live in fixtures.clj. I was not totally sure about the approach and kept the implementation as isolated as possible.

Reply:
> The thought was to prepare the dataset, write on test for existing maps, and then clean up the data after the test is done. 

The cleanup is done by the *outer* database connection and system fixture, right?

I also don't have a strong opinion but is *different* than the rest of functional tests

Reply:
It seems to be that the joggling with those [DDL](https://www.postgresql.org/docs/9.5/ddl.html) is already handled by migrating/rollback approach